oauthenticator/openshift.py

```diff
@@ -13,11 +13,15 @@
 
 
 class OpenShiftOAuthenticator(OAuthenticator):
-    login_service = "OpenShift"
+    user_auth_state_key = "openshift_user"
 
-    scope = ['user:info']
+    @default("scope")
+    def _scope_default(self):
+        return ["user:info"]
 
-    user_auth_state_key = "openshift_user"
+    @default("login_service")
+    def _login_service_default(self):
+        return os.environ.get("LOGIN_SERVICE", "OpenShift")
```

Comment:
Wondering why a dynamic default here? Do we wish this to be configurable?

Reply:
It is configurable in Authenticator, then it was overridden to a string constant in OAuthenticator, and then a few but not all classes made it configurable again.

I'd love to deprecate a few environment variables used to configure things, but I figure its out of scope for this PR trying to be documentation focused.

Reply:
Ah, so CUnicode is something casted to a string. So, giving it the intger will cast it to a string. So instead of `Unicode` erroring if its set with `4`, it would instead work out and have it casted to `"4"`.

Reply:
There is also `CInt` for example, which then would cast to an Int.